contracts/token/ERC721/Tatum721.sol

```diff
@@ -53,6 +54,23 @@ contract Tatum721 is ERC721Enumerable, ERC721URIStorage, AccessControlEnumerable
         return true;
     }
 
+    function mintWithCashback(address to, uint256 tokenId, string memory tokenURI, address[] authorAddresses, uint256[] cashbackValues) public returns (bool) {
+        require(hasRole(MINTER_ROLE, _msgSender()), "ERC721PresetMinterPauserAutoId: must have minter role to mint");
+        _mint(to, tokenId);
+        for (uint i=0;i<authorAddresses.length;i++){
+            _cashbacks[tokenId] = to;
+            _cashbacksValue[tokenId]=cashbackValues[i];
+            safeTransfer(authorAddresses[i],tokenId);    
+        }    
+    }
+    function transferWithCashback(address to, uint256 tokenId) public payable {
+        require(_cashbacks[tokenId] != address(0), "CashbackToken should be of cashback type");
+        require(_cashbacksValue[tokenId] <= msg.value, "Value should be greater than or equal to cashback value");
+        _cashbacks[tokenId] = msg.sender;
```

Comment:
this is wrong. values from cashbacks and cashbacksValue are read only here and used only for finding a recipient of the cashback. Logic here should be like this.
`for (uint i=0;i<_cashbacks[tokenId].length;i++){
_cashbacks[tokenId][i].send(_cashbacksValue[tokenId][i])
}`